src/test/java/underlay/UnderlayTest.java

```diff
@@ -0,0 +1,90 @@
+package underlay;
+
+import blockchain.LightChainRMIInterface;
+import blockchain.Transaction;
+import org.junit.jupiter.api.Test;
+import signature.SignedBytes;
+import skipGraph.NodeInfo;
+import skipGraph.RMIInterface;
+import underlay.requests.GetLeftNodeRequest;
+import underlay.requests.GetPublicKeyRequest;
+import underlay.requests.GetRightNumIDRequest;
+import underlay.requests.PoVRequest;
+import underlay.responses.NodeInfoResponse;
+import underlay.responses.IntegerResponse;
+import underlay.responses.PublicKeyResponse;
+import underlay.responses.SignatureResponse;
+
+import java.io.FileNotFoundException;
+import java.rmi.RemoteException;
+import java.security.PublicKey;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+
+/**
+ * Underlay Unit Tests to make sure messages are sent/received correctly.
+ */
+public class UnderlayTest {
+    private final Underlay underlay = new Underlay();
+
+    @Test
+    public void RightNumIDTest() throws RemoteException, FileNotFoundException {
+        RMIInterface targetRMI = mock(RMIInterface.class);
+        underlay.setTargetRMI(targetRMI);          // method created just for testing purposes
+        when(targetRMI.getRightNumID(0,60)).thenReturn(80);
+        assertEquals(((IntegerResponse) underlay.sendMessage(new GetRightNumIDRequest(0,60), "test")).result
+                , 80);
+    }
+
+    @Test
+    public void GetLeftNodeTest() throws RemoteException, FileNotFoundException {
+        RMIInterface targetRMI = mock(RMIInterface.class);
+        underlay.setTargetRMI(targetRMI);          // method created just for testing purposes
+        NodeInfo result = new NodeInfo("1.1.1.1", 12, "test");
```

Comment:
This sort of object creation during the test is called _test fixturing_. As a best practice, we should not have fixture creating detail during our test. Instead, we better to have a `NodeInfoFixture` in our `src/text/java/fixture` class that creates _randomly_ generated fixtures for our tests. The fixtures should be meaningful, i.e., we expect the first parameter of `NodeInfo` constructor to be a valid IP address. Hence we better to implement `NodeInfoFixture` such that it generates a valid but random IP address for the node, as well as its other info. 

Then we can have `NodeInfo result = NodeInfoFixture()`. 

Please apply this to all tests in this test class, e.g., `TransactionFixture`, `SingedBytesFixture`, etc.